### PR TITLE
[test/multikueue] Move kubecofig generation in the go code.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Move the multikueue kubeconfig file generation form the hack shell scripts into the multikueue e2e go code.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/e2e/multikueue/suite_test.go [hunk 1]
```diff
@@ -47,6 +56,111 @@ var (
...
 
+func policyRule(group, resource string, verbs ...string) rbacv1.PolicyRule {
+	return rbacv1.PolicyRule{
+		APIGroups: []string{group},
+		Resources: []string{resource},
+		Verbs:     verbs,
+	}
+}
+
+func kubeconfigForMultiKueueSA(ctx context.Context, c client.Client, restConfig *rest.Config, ns string, prefix string, clusterName string) ([]byte, error) {
```
Comment (alculquicondor) [inline test/e2e/multikueue/suite_test.go:67]: Let's take a chance to add a comment on this function for all the things we are creating here.

**Review by alculquicondor — COMMENTED** [review]
/approve

@mbobrovskyi can you confirm that this setup runs on OS X?

**Comment by alculquicondor** [comment]
It looks like there is a new flaky

**Comment by alculquicondor** [comment]
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/2526/pull-kueue-test-e2e-main-1-30/1808529251364769792

**Comment by mbobrovskyi** [comment]
@alculquicondor yes, on mac os working fine.

**Comment by alculquicondor** [comment]
/approve

Please add the comment to document what is required to make the kubeconfig.

File: test/e2e/multikueue/suite_test.go (hunk 1 above)
Comment (trasc) [reply test/e2e/multikueue/suite_test.go:77]: done

**Comment by trasc** [comment]
> /approve
> 
> Please add the comment to document what is required to make the kubeconfig.

done

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Pull request merged**